File: app/src/main/java/memoque/bobs/com/memoque/main/adapter/MemoQueAdapter.java
package memoque.bobs.com.memoque.main.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import memoque.bobs.com.memoque.R;
import memoque.bobs.com.memoque.main.MemoQueManager;
import memoque.bobs.com.memoque.main.adapter.MemoQueAdapter.ViewHolder;
import memoque.bobs.com.memoque.main.memo.MemoModel;

public class MemoQueAdapter extends RecyclerView.Adapter<ViewHolder>
{
	private List<MemoModel> memoModelList;

	public MemoQueAdapter()
	{
		memoModelList = MemoQueManager.Companion.getInstance().getMemoList();
	}

	@NonNull
	@Override
	public ViewHolder onCreateViewHolder( @NonNull ViewGroup viewGroup, int i )
	{
		CardView cardView = (CardView)LayoutInflater.from( viewGroup.getContext() ).inflate( R.layout.recycler_view_row, viewGroup, false );
		return new ViewHolder( cardView );
	}

	@Override
	public void onBindViewHolder( @NonNull ViewHolder viewHolder, int i )
	{
		CardView cardView = viewHolder.cardView;
		MemoModel memoData = memoModelList.get( i );

		TextView title = cardView.findViewById( R.id.memo_title );
		title.setText( memoData.getTitle() );

		TextView content = cardView.findViewById( R.id.memo_content );
		content.setText( memoData.getContent() );

		TextView date = cardView.findViewById( R.id.memo_date );
		date.setText( memoData.getDate() );
	}

	@Override
	public int getItemCount()
	{
		if( null == memoModelList )
			return 0;

		return memoModelList.size();
	}

	static class ViewHolder extends RecyclerView.ViewHolder
	{
		private CardView cardView;

		ViewHolder( CardView v )
		{
			super( v );
			cardView = v;
		}
	}
}

File: app/src/main/java/memoque/bobs/com/memoque/main/adapter/MemoQueAdapter.kt
package memoque.bobs.com.memoque.main.adapter

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import memoque.bobs.com.memoque.R
import memoque.bobs.com.memoque.appdata.AppData
import memoque.bobs.com.memoque.main.MemoQueManager
import memoque.bobs.com.memoque.main.adapter.MemoQueAdapter.ViewHolder
import memoque.bobs.com.memoque.main.memo.BSMemo
import memoque.bobs.com.memoque.main.memo.DetailMemoActivity

open class MemoQueAdapter : RecyclerView.Adapter<ViewHolder>(), IAdapter {

    protected var BSMemoList: List<BSMemo>? = null

    open fun initData() {
        // 매니져에 어뎁터를 세팅하고 메모리스트를 가져온다
        MemoQueManager.instance.setAdapterListener(MemoQueManager.Adapterkey.MEMO, this)
        BSMemoList = MemoQueManager.instance.getMemos()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val cardView = LayoutInflater.from(viewGroup.context).inflate(R.layout.recycler_view_row, viewGroup, false) as CardView
        return ViewHolder(cardView)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        // 리사이클러뷰 카드뷰별 세팅
        val cardView = viewHolder.cardView
        val memo = BSMemoList!![i]

        val title = cardView.findViewById<TextView>(R.id.memo_title)
        title.text = memo.title

        val content = cardView.findViewById<TextView>(R.id.memo_content)
        content.text = memo.content

        val date = cardView.findViewById<TextView>(R.id.memo_date)
        date.text = memo.date

        cardView.setOnClickListener { run {
            val intent = Intent(AppData.mainActivity, DetailMemoActivity::class.java)
            intent.putExtra(DetailMemoActivity.MEMO_INDEX, memo.index)
            intent.putExtra(DetailMemoActivity.TAB_KEY, getAdapterKey())
            AppData.mainActivity.startActivity(intent)
        } }
    }

    open fun getAdapterKey() : MemoQueManager.Adapterkey{
        return MemoQueManager.Adapterkey.MEMO
    }

    override fun getItemCount(): Int {
        // 리사이클러뷰 카드뷰 개수 리턴
        return if (null == BSMemoList) 0 else BSMemoList!!.size
    }

    class ViewHolder(val cardView: CardView) : RecyclerView.ViewHolder(cardView)

    override fun refreshAll() {
        // 전체 메모 리스트 카드뷰 갱신
        BSMemoList = MemoQueManager.instance.getMemos()
        notifyDataSetChanged()
    }

    override fun searchMemos(BSMemos: List<BSMemo>?, searchtext: String) {
        // 검색탭 메모 리스트 갱신
        BSMemoList = BSMemos
        notifyDataSetChanged()
    }

    override fun clear() {
        if (BSMemoList == null || BSMemoList!!.isEmpty())
            return

        BSMemoList = mutableListOf()
        notifyDataSetChanged()
    }
}
